step-capstone/src/components/Finalized.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import {useEffect, useState, useRef} from 'react';
+import ReactDOM from 'react-dom';
+import TravelObject from './TravelObject';
+import TimeLine from './TimeLine';
+export default function Finalized(props) {
+    const hashMap = new Map();
+    let prevDiv = useRef([]);
+    const [displayDate, setDisplayDate] = useState(props.startDate);
+
+    const handleDisplayDateChange = (newDisplayDate) => {
+        setDisplayDate(newDisplayDate);
+    }
+
+    const sameDate = (timeA, timeB) => {
+      return (timeA.getDate() === timeB.getDate() && timeA.getMonth() === timeB.getMonth() && timeA.getFullYear() === timeB.getFullYear());
+    }
+    const pushItemToArrayInHashMap = (item, dateKey) => {
+        if (!hashMap.has(dateKey)) {
+            hashMap.set(dateKey, [item]);
+        }
+        else {
+            let array = hashMap.get(dateKey);
+            array.push(item);
+            hashMap.set(dateKey, array);
+        }
+    }
+    const separateDates = () => {
+        for(let i = 0; i < props.list.length; i++) {
+            let item = props.list[i];
+            let startDate = item.startDate;
+            let endDate = item.endDate;
+            pushItemToArrayInHashMap(item, startDate.toDateString());
+            if (!sameDate(startDate, endDate)) {
+                pushItemToArrayInHashMap(item, endDate.toDateString());
+            }
+        }
+    }
+    const travelObjectstartDateComparator = (travelObjectA, travelObjectB) => {
+        if (travelObjectA.startDate ===  travelObjectB.startDate) {
+            return 0;
+        }
+        else if (travelObjectA.startDate >  travelObjectB.startDate) {
+            return 1;
+        }
+        else   {
+            return -1;
+        }
+    }
+    const sortHashMap = () => {
+        const iterator = hashMap[Symbol.iterator]();
+        for (let KV of iterator) {
+            let V = KV[1];
+            V.sort(travelObjectstartDateComparator);
+        }
+    }
+    
+    useEffect(() => {
+        setDisplayDate(props.startDate);
+    }, [props.startDate]);
+    
+    const getAssociatedDiv = (startDate) => {
```

Comment:
I'm a bit confused as to what this does/what it's for?

Reply:
is it its place in the itinerary?

Reply:
Yes, this function gets the div that the TravelObject clings to.

Reply:
Let me add comments to the function to clarify